### PR TITLE
Alt-Az system integrated with boost::units

### LINKED ISSUE #53: Integrate alt_az with units
Updating alt-az to accept the components with boost::units

### PR DESCRIPTION
**Pull request by lpranam**
### Description
`coordinate/alt_az.hpp` updated to work with unit integrated geometric coordinate system.

### References
closes #53

### Tasklist

- [ ] Add test case(s)
- [x] Ensure all CI builds pass
- [x] Review and approve

### CODE REVIEW

File: include/boost/astronomy/coordinate/alt_az.hpp [hunk 1]
```diff
@@ -19,248 +23,359 @@
...
+        Differential::quantity1 const& pm_alt,
+        Differential::quantity2 const& pm_az_cosalt,
+        Differential::quantity3 const& radial_velocity) :
+        alt_az(alt, az, distance)
+    {
+        this->motion.set_dlat_dlon_coslat_ddist(pm_alt, pm_az_cosalt, radial_velocity);
+    }
+
+    template <typename OtherRepresentation, typename OtherDifferential>
+    alt_az(Representation const& representation_data, Differential const& diff)
```
Comment (Ruchika2696) [inline include/boost/astronomy/coordinate/alt_az.hpp]: parameters should be changed to `OtherRepresentation` and `OtherDifferential`

**Review by Ruchika2696 — CHANGES_REQUESTED** [review]

**Comment by Ruchika2696** [comment]
missing copy constructor?

**Review by Ruchika2696 — CHANGES_REQUESTED** [review]
missing copy constructor!!

File: include/boost/astronomy/coordinate/alt_az.hpp [hunk 2]
```diff
@@ -19,248 +23,370 @@
...
+        Differential::quantity1 const& pm_alt,
+        Differential::quantity2 const& pm_az_cosalt,
+        Differential::quantity3 const& radial_velocity) :
+        alt_az(alt, az, distance)
+    {
+        this->motion.set_dlat_dlon_coslat_ddist(pm_alt, pm_az_cosalt, radial_velocity);
+    }
+
+    template <typename OtherRepresentation, typename OtherDifferential>
+    alt_az(OtherRepresentation const& representation_data, OtherDifferential const& diff)
```
Comment (sarthak2007) [inline include/boost/astronomy/coordinate/alt_az.hpp]: Change `diff` to `differential_data` as you used `differential_data` below

**Review by sarthak2007 — CHANGES_REQUESTED** [review]

**Review by Ruchika2696 — APPROVED** [review]

**Review by sarthak2007 — APPROVED** [review]

**Pull request merged**